OMEdit/OMEditLIB/Modeling/ModelWidgetContainer.cpp

```diff
@@ -5415,7 +5408,8 @@ void ModelWidget::updateChildClasses(LibraryTreeItem *pLibraryTreeItem)
  */
 bool ModelWidget::omsimulatorEditorTextChanged()
 {
-  if (OMSProxy::instance()->loadSnapshot(mpLibraryTreeItem->getNameStructure(), mpEditor->getPlainTextEdit()->toPlainText())) {
+  QString newCref;
+  if (OMSProxy::instance()->importSnapshot(mpLibraryTreeItem->getNameStructure(), mpEditor->getPlainTextEdit()->toPlainText(), &newCref)) {
```

Comment:
@adeas31 `importSnapshot` is now taking the model name as first argument, instead of the cref. I don't know how to change it here.

Reply:
So you mean that it only works for top level model, right?

Reply:
No, it works on all levels. But the path is stored within the snapshot, and the first argument should just be the model name.